pymodbus/client/base.py

```diff
@@ -216,7 +216,7 @@ def build_response(self, tid):
     # ----------------------------------------------------------------------- #
     # Internal methods
     # ----------------------------------------------------------------------- #
-    def send(self, request):
+    def send(self, request) -> int:  # type: ignore [empty-body]
```

Comment:
Avoid the ignore by adding a dummy return 0, please.

Reply:
I do wonder why it does not complain on line 225-229, it is exactly the same issue (recv returns bytes)...that just goes to show how (sorry for the expression) how incompetent and dangerous these checkers can be :-)

Reply:
`mypy` is not currently checking in CI since the entire definition is un-typed.  It complains if using [`--check-untyped-defs`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-check-untyped-defs).  (`pyright` is even stricter/smarter and complains about things `mypy` misses).

My intention with these PRs is to reduce the number of errors with `--check-untyped-defs` down to zero so we can turn on that feature in `pyproject.toml`.  (It started > 200 and is now ~80)

Reply:
You are doing a great job !!! and I try to faclitate it as much as I can, while using my experience to try and prevent things than can cause bigger problems.

You are running point on these changes, and I accept that our code becomes more C++ like everyday....no harm I actually prefer coding C++.

Reply:
> Avoid the ignore by adding a dummy return 0, please.

I prefer to keep the type error for any calling code which tries to call `ModbusBaseClient.send()` instead of one of the subclasses.

Reply:
Actually, using `AbstractBaseClass` is even better since nobody should ever call this `ModbusBaseClient.send()`.  I tried this and got stuck since many of the tests use it. :(

Reply:
> You are running point on these changes, and I accept that our code becomes more C++ like everyday....no harm I actually prefer coding C++.
![image](https://github.com/pymodbus-dev/pymodbus/assets/52292902/a2574af6-b9f5-4fea-8383-9f04f6b16fcb)

:)

Reply:
Make an issue regarding abstract and let's get this merged.

Reply:
https://github.com/pymodbus-dev/pymodbus/issues/2036